panc/src/main/scripts/panlint.py

```diff
@@ -0,0 +1,307 @@
+                    end_line = start_line + raw_text[start_char:end_char].count('\n')
+                    for i in range(start_line, end_line + 1):
+                        ignore_lines.append(i)
+
+                for line_number, line in enumerate(raw_text.splitlines(), start=1):
+                    line = line.rstrip('\n')
+
+                    if line and line_number not in ignore_lines and not RE_COMMENT_LINE.match(line):
+                        debug_line(line, line_number)
+                        if first_line:
+                            first_line = False
+                            if not RE_FIRST_LINE.match(line):
+                                print_fileinfo(filename, line_number, 'First non-comment line must be the template type and name')
+                                print_line(line)
+                                print_diagnosis(diagnose(0, len(line)))
+
+                        else:
+                            messages = []
+                            diagnoses = []
+
+                            line = RE_TRAILING_COMMENT.sub('', line)
+
+                            string_ranges = get_string_ranges(line)
+
+                            for message, pattern in LINE_PATTERNS.iteritems():
+                                m = pattern.search(line)
+                                if m and m.group('error'):
+                                    start, end = m.span('error')
+                                    debug_range(start, end, 'Match', True)
+                                    if not inside_string(start, end, string_ranges):
+                                        diagnoses.append(diagnose(start, end))
+                                        messages.append(message)
+                                        problems_found += 1
+                                        problem_stats[filename] += 1
+
+                            for _, check_method in getmembers(line_checks, predicate=ismethod):
+                                passed, diagnosis, message = check_method(line, string_ranges)
+                                if not passed:
+                                    diagnoses.append(diagnosis)
+                                    messages.append(message)
+                                    problems_found += 1
+                                    problem_stats[filename] += 1
+
+                            if messages and diagnoses:
+                                reports.append([filename, line_number, line, merge_diagnoses(diagnoses), ', '.join(messages)])
+                    else:
+                        debug_ignored_line(line, line_number)
+
+
+        for report in reports:
+            print_report(*report, vi=args.vi)
+
+        if args.table:
+            print
+            print 'Problem count per file:'
+            print filestats_table(problem_stats)
+
+        print
+        print '%d problems found in total' % problems_found
+
+        if problems_found:
```

Comment:
I wouldn't recommend returning `problems_found` in that case as the exit status must be between 0 and 255.